R/service-azure_openai.R

```diff
@@ -110,31 +110,33 @@ query_api_azure_openai <-
 retrieve_azure_token <- function() {
   rlang::check_installed("AzureRMR")
 
-  token <- tryCatch(
-    {
-      AzureRMR::get_azure_login(
-        tenant = Sys.getenv("AZURE_OPENAI_TENANT_ID"),
-        app = Sys.getenv("AZURE_OPENAI_CLIENT_ID"),
-        scopes = ".default"
-      )
-    },
-    error = function(e) NULL
-  )
+  token <- retrieve_azure_token_object() %>% suppressMessages()
 
-  if (is.null(token)) {
-    token <- AzureRMR::create_azure_login(
-      tenant = Sys.getenv("AZURE_OPENAI_TENANT_ID"),
-      app = Sys.getenv("AZURE_OPENAI_CLIENT_ID"),
-      password = Sys.getenv("AZURE_OPENAI_CLIENT_SECRET"),
-      host = "https://cognitiveservices.azure.com/",
-      scopes = ".default"
-    )
-  }
+  invisible(token$credentials$access_token)
+}
+
+
+retrieve_azure_token_object <- function() {
+  ## Set this so that do_login properly caches
+  azure_data_env = Sys.getenv("R_AZURE_DATA_DIR")
+  Sys.setenv("R_AZURE_DATA_DIR" = gptstudio_cache_directory())
 
-  invisible(token$token$credentials$access_token)
+  client <- Microsoft365R:::do_login(tenant = Sys.getenv("AZURE_OPENAI_TENANT_ID"),
```

Comment:
CRAN might yell about us depending on a hidden function, but I'm not sure what the rules are.

Reply:
On these situations I would prefer to reverse-engineer the function to have a local version instead of a dependency. But the level of effort required varies between cases

Reply:
Agreed. I did that, removing the Microsoft365R dependency, adding AzureGraph to Suggested list, and adding the `rlang::check_installed("AzureGraph")` in the function. So if it isnt installed, it will be prompted.